src/main/java/com/aws/greengrass/easysetup/GreengrassSetup.java

```diff
@@ -510,9 +517,20 @@ private String peekArg() {
 
     void provision(Kernel kernel) throws IOException, DeviceConfigurationException {
         outStream.printf("Provisioning AWS IoT resources for the device with IoT Thing Name: [%s]...%n", thingName);
-        final ThingInfo thingInfo =
-                deviceProvisioningHelper.createThing(deviceProvisioningHelper.getIotClient(), thingPolicyName,
-                        thingName);
+        // handle endpoints provided by external config
+        String iotDataEndpoint = "";
+        String iotCredEndpoint = "";
+        if (Coerce.toBoolean(kernel.getConfig().find(SERVICES_NAMESPACE_TOPIC,
+                DEFAULT_NUCLEUS_COMPONENT_NAME, CONFIGURATION_CONFIG_KEY, DEVICE_PARAM_FIPS_MODE))) {
+            iotDataEndpoint  = Coerce.toString(kernel.getConfig().find(SERVICES_NAMESPACE_TOPIC,
+                    DEFAULT_NUCLEUS_COMPONENT_NAME, CONFIGURATION_CONFIG_KEY, DEVICE_PARAM_IOT_DATA_ENDPOINT));
+            iotCredEndpoint = Coerce.toString(kernel.getConfig().find(SERVICES_NAMESPACE_TOPIC,
+                    DEFAULT_NUCLEUS_COMPONENT_NAME, CONFIGURATION_CONFIG_KEY, DEVICE_PARAM_IOT_CRED_ENDPOINT));
+        }
+
```

Comment:
I agree with your previous code to update the endpoints only if those are not provided irrespective of the fips flag. 

Here, you can use findOrDefault with default making DescribeEndpointRequest

Reply:
Sure, I prefer to change it back, because all requests made through iot client are handled in `DeviceProvisioningHelper` and `GreengrassSetup` handles configs with kernel. I'm a bit hesitate to inject kernel into `DeviceProvisioningHelper`

Reply:
Updated